network/discovery/dv5_service.go

```diff
@@ -304,15 +304,15 @@ func (dvs *DiscV5Service) publishENR() {
 				// ignore
 				return
 			}
-			dvs.logger.Warn("could not ping node", zap.String("targetNodeENR", e.Node.String()), zap.Error(err))
+			logger.Warn("could not ping node", zap.String("targetNodeENR", e.Node.String()), zap.Error(err))
 			return
 		}
 		metricPublishEnrPongs.Inc()
-		// dvs.logger.Debug("ping success", zap.String("targetNodeENR", e.Node.String()))
-	}, time.Millisecond*100, dvs.badNodeFilter)
+		// logger.Debug("ping success", zap.String("targetNodeENR", e.Node.String()))
+	}, time.Millisecond*100, dvs.badNodeFilter(logger))
```

Comment:
oh, i see. because we need a func to pass here

Reply:
yeah